Jenkinsfile

```diff
@@ -1,52 +1,123 @@
-    }
-    stage('Tests') {
-      steps {
-        runXmostest("${REPO}", 'tests')
+      options {
+        skipDefaultCheckout()
       }
-    }
-    stage('xCORE builds') {
-      steps {
-        dir("${REPO}") {
-          xcoreAllAppsBuild('examples')
-          xcoreAllAppNotesBuild('examples')
-          dir("${REPO}") {
-            runXdoc('doc')
+      stages{
+        stage('Get view') {
+          steps {
+            xcorePrepareSandbox("${VIEW}", "${REPO}")
+          }
+        }
+        stage('Library checks') {
+          steps {
+            xcoreLibraryChecks("${REPO}")
+          }
+        }
+        stage('Tests') {
+          steps {
+            runXmostest("${REPO}", 'tests')
+          }
+        }
+        stage('xCORE builds') {
+          steps {
+            dir("${REPO}") {
+              xcoreAllAppsBuild('examples')
+              xcoreAllAppNotesBuild('examples')
+              dir("${REPO}") {
+                runXdoc('doc')
+              }
+              //Build these individually (or we can extend xcoreAllAppsBuild to support an argument
+              dir('examples/app_debug_printf'){
+                runXmake(".", "", "XCOREAI=1")
+                stash name: 'app_debug_printf', includes: 'bin/xcoreai/*.xe, '
+              }
+              dir('examples/AN00239'){
+                runXmake(".", "", "XCOREAI=1")
+                stash name: 'AN00239', includes: 'bin/xcoreai/*.xe'
+              }
+              dir('tests/debug_printf_test'){
+                runXmake(".", "", "XCOREAI=1")
+                stash name: 'debug_printf_test', includes: 'bin/xcoreai/*.xe'
+              }
+            }
+          }
+        }
+      }// stages
+      post {
+        cleanup {
+          xcoreCleanSandbox()
+        }
+      }
+    }// Stage standard build
+
+    stage('xcore.ai Verification'){
+      agent {
+        label 'xcore.ai-explorer'
+      }
+      stages{
+        stage('Install Dependencies') {
+          steps {
+            sh '/XMOS/get_tools.py ' + params.TOOLS_VERSION
+            installDependencies()
+          }
+        }
+        stage('xrun'){
+          steps{
+            toolsEnv(TOOLS_PATH) {  // load xmos tools
+              //Run this and diff against expected output. Note we have the lib files here available
+              unstash 'debug_printf_test'
+              sh 'xrun --io --id 0 bin/xcoreai/debug_printf_test.xe &> debug_printf_test.txt'
+              sh 'cat debug_printf_test.txt && diff debug_printf_test.txt tests/test.expect'
+
+              //Just run these and error on exception
+              unstash 'AN00239'
+              sh 'xrun --io --id 0 bin/xcoreai/AN00239.xe'
+              unstash 'app_debug_printf'
+              sh 'xrun --io --id 0 bin/xcoreai/app_debug_printf.xe'
+            }
           }
         }
+      }//stages
+      post {
+        cleanup {
+          cleanWs()
+        }
+      }
+    }// xcore.ai
+
+    stage('Update view files') {
```

Comment:
Without this inside a post block, it will run in all cases. Usually a build failure will stop the build but could be skipped with `warnError` or `catchError` which continues the build whilst still marking it as failed. This would for example update the viewfiles even if library checks failed as they are wrapped in `warnError`.

Options could be:
- go back to the `node` syntax but also include a stage so it appears on the UI (I'm guessing thats why you went with declarative here). 
- Add a check for build success using `currentBuild.currentResult` or something similar